graphql/schemabuilder/reflect.go

```diff
@@ -350,6 +355,12 @@ func (sb *schemaBuilder) makeStructParser(typ reflect.Type) (*argParser, graphql
 		return nil, nil, fmt.Errorf("expected struct but received type %s", typ.Name())
 	}
 
+	// Cache type information ahead of time to catch self-reference
+	if sb.typeCache == nil {
+		sb.typeCache = make(map[reflect.Type]cachedType, 0)
+	}
```

Comment:
why do we need to check this here if it's already initialized by https://github.com/samsarahq/thunder/pull/120/files#diff-14d89fd44582e138b4a541885877e741R1155 ?